src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormat.cs

```diff
@@ -657,9 +657,8 @@ private static bool IsUseGenitiveForm(ReadOnlySpan<char> format, int index, int
                         if (isJapaneseCalendar &&
                             !AppContextSwitches.FormatJapaneseFirstYearAsANumber &&
                             year == 1 &&
-                            i + tokenLen < format.Length - 1 &&
-                            format[i + tokenLen] == '\'' &&
-                            format[i + tokenLen + 1] == DateTimeFormatInfoScanner.CJKYearSuff[0])
+                            ((i + tokenLen < format.Length && format[i + tokenLen] == DateTimeFormatInfoScanner.CJKYearSuff[0]) ||
+                            (i + tokenLen < format.Length - 1 && format[i + tokenLen] == '\'' && format[i + tokenLen + 1] == DateTimeFormatInfoScanner.CJKYearSuff[0])))
```

Comment:
Is this going to cause us to misinterpret "malformed" input on Windows whereas we wouldn't have previously?  Or we're basically just ok allowing either format on either platform?

Reply:
We allow both which is more desirable.